### ISSUE TITLE
[BUG] ZAP tool --> Primary Device is not correctly saved to zap file

### ISSUE THREAD

**Issue by lboue** [project-chip/zap#1420]
### Reproduction steps

I am facing an issue with ZAP tool. Primary Device is not correctly saved to zap file. For example, with PR #35410.

**Reproduce steps**

- `cd ~/connectedhomeip`
- `./scripts/tools/zap/run_zaptool.sh examples/chef/devices/rootnode_waterleakdetector_0b067acfa3.zap`
- Edit EP1 and add `Matter Power Source` in the device list
- Check that Primary Device is still `Matter Water Leak Detector`
- Save save with CTRL+S
- Close the ZCL windows
- `./scripts/tools/zap/run_zaptool.sh examples/chef/devices/rootnode_waterleakdetector_0b067acfa3.zap`
- Edit EP1 and notice that Primary Device is not `Matter Water Leak Detector` but `Matter Power Source`

### Bug prevalence

Always

### GitHub hash of the SDK that was being used

c6f66811c00f6c13e09938147a7c37310ee7bdc9

### Platform

core

### Platform Version(s)

N/A

**Comment by lboue** [comment]
Same issue: [Not able to set SmokeAlarmCO as Primary device type if PowerSource device type exists #1119](https://github.com/project-chip/zap/issues/1119)

**Comment by lboue** [comment]
Fixed by [v2024.09.10-nightly](https://github.com/project-chip/zap/releases/tag/v2024.09.10-nightly) release.

**Testing**
```
./scripts/tools/zap/zap_download.py --zap-version v2024.09.10-nightly
./scripts/tools/zap/run_zaptool.sh examples/chef/devices/rootnode_waterleakdetector_0b067acfa3.zap
```
![image](https://github.com/user-attachments/assets/bb6f3b9f-330a-4dc8-bdff-c44768ce01bf)
